rust/connlib/tunnel/src/device_channel/tun_linux.rs

```diff
@@ -355,11 +357,33 @@ async fn configure_network_manager(_dns_config: &[IpAddr]) -> Result<()> {
     ))
 }
 
-async fn configure_systemd_resolved(_dns_config: &[IpAddr]) -> Result<()> {
-    // TODO: Call `resolvectl` here
-    Err(Error::Other(
-        "DNS control with `systemd-resolved` is not implemented yet",
-    ))
+async fn configure_systemd_resolved(dns_config: &[IpAddr]) -> Result<()> {
+    let mut cmd = tokio::process::Command::new("resolvectl");
+    cmd.arg("dns");
+    cmd.arg(IFACE_NAME);
+    dns_config.iter().for_each(|addr| {
+        cmd.arg(addr.to_string());
+    });
+    let status = cmd
+        .status()
+        .await
+        .map_err(|_| Error::Other("`resolvectl dns` should have run"))?;
```

Comment:
```suggestion
    let status = tokio::process::Command::new("resolvectl")
        .arg("dns")
        .arg(IFACE_NAME)
        .args(dns_config.iter().map(ToString::to_string))
        .status()
        .await
        .map_err(|_| Error::Other("`resolvectl dns` should have run"))?;
```

Also, I think it'll be better if we create a new error type instead of using `Other`.